### PR TITLE
3.1.x procs oob remove

### PR DESCRIPTION
**Pull request by artpol84**

### CODE REVIEW

File: orte/mca/ess/base/ess_base_std_app.c [hunk 1]
```diff
@@ -308,17 +194,6 @@ int orte_ess_base_app_setup(bool db_restrict_local)
         error = "orte_cr_init";
         goto error;
     }
-    /* open the distributed file system */
```
Comment (artpol84) [inline orte/mca/ess/base/ess_base_std_app.c]: @karasevb Why this needs to be removed?

**Comment by artpol84** [comment]
This is just to run this through the CI. Eventually it will go into the master.

**Comment by rhc54** [comment]
FWIW: I have a branch that includes the app framework updates that is in test now. There is more required than is included here.

**Comment by artpol84** [comment]
@rhc54 are you going to open a PR sometime soon?
Can you share what are other concerns? Dynamics?

**Comment by rhc54** [comment]
Yeah, probably in the next day or so. It can't go into v3.1.0 anyway, so we have time.

**Comment by artpol84** [comment]
I need to isolate what is needed to remove those components from 3,1 as I may need a custom built.

**Comment by rhc54** [comment]
ok, well I need to finish something else first, but will post a PR as soon as I can.

**Comment by artpol84** [comment]
thank you

**Comment by artpol84** [comment]
@karasevb 
no need to close this frameworks:
https://github.com/open-mpi/ompi/pull/4986/commits/eb556780c92b33b4a31753a2b31e0f48942cc32b#diff-fe9ff9b2652b1c108b8a0a1bd1fd3fffL348

**Comment by karasevb** [comment]
Here are the results of MTT for this PR:
```
+-------------+--------------------------------+------------------------+----------+------+------+----------+------+---------------------------------------------------------------------------------------------------+
| Phase       | Section                        | MPI Version            | Duration | Pass | Fail | Time out | Skip | Detailed report                                                                                   |
+-------------+--------------------------------+------------------------+----------+------+------+----------+------+---------------------------------------------------------------------------------------------------+
| MPI Install | ompi_vendor_debug              | 3.1.x_procs_oob_remove | 21:49    | 1    |      |          |      | MPI_Install-ompi_vendor_debug-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html           |
| Test Build  | mpi-test-suite                 | 3.1.x_procs_oob_remove | 00:31    | 1    |      |          |      | Test_Build-mpi-test-suite-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html               |
| Test Build  | imb                            | 3.1.x_procs_oob_remove | 00:02    | 1    |      |          |      | Test_Build-imb-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                          |
| Test Build  | imb-datacheck                  | 3.1.x_procs_oob_remove | 00:02    | 1    |      |          |      | Test_Build-imb-datacheck-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                |
| Test Build  | osu_micro_benchmark            | 3.1.x_procs_oob_remove | 00:27    | 1    |      |          |      | Test_Build-osu_micro_benchmark-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html          |
| Test Build  | mpich_tests                    | 3.1.x_procs_oob_remove | 10:05    | 1    |      |          |      | Test_Build-mpich_tests-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                  |
| Test Build  | intel                          | 3.1.x_procs_oob_remove | 00:47    | 1    |      |          |      | Test_Build-intel-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                        |
| Test Build  | hash                           | 3.1.x_procs_oob_remove | 00:03    | 1    |      |          |      | Test_Build-hash-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                         |
| Test Build  | atest                          | 3.1.x_procs_oob_remove | 00:07    | 1    |      |          |      | Test_Build-atest-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                        |
| Test Build  | dummy                          | 3.1.x_procs_oob_remove | 00:00    | 1    |      |          |      | Test_Build-dummy-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                        |
| Test Build  | trivial                        | 3.1.x_procs_oob_remove | 00:01    | 1    |      |          |      | Test_Build-trivial-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                      |
| Test Run    | imb-datacheck                  | 3.1.x_procs_oob_remove | 02:03:00 | 12   |      |          |      | Test_Run-imb-datacheck-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                  |
| Test Run    | mpich_tests_mpi_datatype       | 3.1.x_procs_oob_remove | 17:25    | 648  |      |          |      | Test_Run-mpich_tests_mpi_datatype-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html       |
| Test Run    | hash                           | 3.1.x_procs_oob_remove | 00:14    | 12   |      |          |      | Test_Run-hash-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                           |
| Test Run    | mpich_tests_mpi_basic          | 3.1.x_procs_oob_remove | 06:50:33 | 84   |      |          |      | Test_Run-mpich_tests_mpi_basic-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html          |
| Test Run    | mpich_tests_mpi_comm           | 3.1.x_procs_oob_remove | 08:27    | 276  |      |          |      | Test_Run-mpich_tests_mpi_comm-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html           |
| Test Run    | osu_micro_benchmark_mbw_mr     | 3.1.x_procs_oob_remove | 00:47    | 12   |      |          |      | Test_Run-osu_micro_benchmark_mbw_mr-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html     |
| Test Run    | mpich_tests_mpi_pt2pt          | 3.1.x_procs_oob_remove | 33:30    | 420  |      |          |      | Test_Run-mpich_tests_mpi_pt2pt-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html          |
| Test Run    | osu_micro_benchmark_collective | 3.1.x_procs_oob_remove | 47:07    | 276  |      |          |      | Test_Run-osu_micro_benchmark_collective-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html |
| Test Run    | imb                            | 3.1.x_procs_oob_remove | 25:09    | 12   |      |          |      | Test_Run-imb-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                            |
| Test Run    | osu_micro_benchmark_one_sided  | 3.1.x_procs_oob_remove | 09:04    | 108  |      |          |      | Test_Run-osu_micro_benchmark_one_sided-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html  |
| Test Run    | intel                          | 3.1.x_procs_oob_remove | 56:45    | 2052 |      |          |      | Test_Run-intel-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                          |
| Test Run    | osu_micro_benchmark_p2p        | 3.1.x_procs_oob_remove | 05:09    | 60   |      |          |      | Test_Run-osu_micro_benchmark_p2p-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html        |
| Test Run    | atest                          | 3.1.x_procs_oob_remove | 44:59    | 180  |      |          |      | Test_Run-atest-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                          |
| Test Run    | trivial                        | 3.1.x_procs_oob_remove | 01:10    | 72   |      |          |      | Test_Run-trivial-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                        |
| Test Run    | mpi-test-suite                 | 3.1.x_procs_oob_remove | 01:57    | 12   |      |          |      | Test_Run-mpi-test-suite-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html                 |
| Test Run    | mpich_tests_mpi_coll           | 3.1.x_procs_oob_remove | 22:36    | 984  |      |          |      | Test_Run-mpich_tests_mpi_coll-ompi_vendor-3.1.x_procs_oob_remove-ompi_vendor_debug.html           |
+-------------+--------------------------------+------------------------+----------+------+------+----------+------+---------------------------------------------------------------------------------------------------+


    Total Tests:    5231
    Total Failures: 0
    Total Passed:   5231
    Total Duration: 50506 secs. (14:01:46)
```

**Comment by jsquyres** [comment]
@artpol84 Is this PR still relevant?  It's marked WIP/DNM, and is for the v3.1.x branch.  Should this cleanup be done on master and eventually naturally make its way to a release branch?

**Comment by artpol84** [comment]
@rhc54 have you finished with the PR you mentioned?

**Comment by bwbarrett** [comment]
@artpol84 and @rhc54 is there any point in continuing with this PR, given the late state of 3.1?

**Comment by artpol84** [comment]
@rhc54 Do you have any updates?
@bwbarrett no rush with 3.1, but I would like to have a vision on how are we going to proceed.

**Comment by rhc54** [comment]
AFAICT, this hasn't been pushed into master yet - and thus shouldn't be proposed for a release branch.

My concern was two fold:

- This PR only removes a portion of the app-related things that are no longer required when using PMIx

- Removing those things may impact others that rely on them but whose code doesn't reside in the OMPI repo. We don't know, for example, if there are academic efforts that would be impacted. At the least, we should provide some warning on the devel mailing list so people can speak up if they object.

I guess I'd like to see us generate a warning and request comments, and then do a complete cleanup on master. Once those things are done, then we should include those changes in the next major release series - we shouldn't remove capabilities mid-series.

Just my $0.02

**Comment by jsquyres** [comment]
Sounds like this should be closed.

If we want this on v3.1, then let's get it in master first.

**Comment by artpol84** [comment]
@jsquyres not sure I understand your conclusion. Can you elaborate?

**Comment by artpol84** [comment]
I don't mind doing some more complex solution that we can discuss next week, but let's not close it as I believe it has a performance-wise value to a jobstart.

**Comment by rhc54** [comment]
Hi @artpol84 - we weren't aware of your status, and noted that the request to check the devel list etc. hadn't moved forward. So I took care of that part, then redirected the PR to OMPI master and found it had gone stale and had a number of conflicts. We therefore did the code reduction in one PR with #5827

**Comment by artpol84** [comment]
Thanks for the reference. Looks like #5827 includes this PR and does some more work (I guess completely removing unused code. Is that correct?

**Comment by rhc54** [comment]
Yes, that's correct - as PMIx replaced functionality, some significant fraction of the ORTE code became expendable.

**Pull request closed without merging**